.github/workflows/release.yaml

```diff
@@ -0,0 +1,152 @@
+name: Release
+on:
+  pull_request:
+    types: [opened, synchronize, reopened, labeled, unlabeled]
+    branches: [main]
+    paths:
+      - "*/Cargo.toml" # only run if Cargo.toml changes
+  pull_request_target:
+    types: [closed]
+    branches: [main]
+    paths:
+      - "*/Cargo.toml" # only run if Cargo.toml changes
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  release:
+    runs-on: ubuntu-latest
+    if: contains(github.event.pull_request.labels.*.name, 'release')
+    env:
+      EARTHLY_TOKEN: "${{ secrets.EARTHLY_TOKEN }}"
+      CARGO_REGISTRY_TOKEN: "${{ secrets.CARGO_REGISTRY_TOKEN }}"
+      FORCE_COLOR: 1
+    steps:
+      - uses: earthly/actions-setup@v1
+        with:
+          version: v0.8.3
+      - uses: actions/checkout@v4
+        with:
+          submodules: true
+      - uses: actions/setup-node@v4
+        with:
+          node-version: '20.x'
+      - run: npm install toml globby axios
+      - uses: actions/github-script@v7
+        with:
+          script: |
+            const is_merge_event = ${{ github.event_name == 'pull_request_target' && github.event.action == 'closed' && github.event.pull_request.merged == true }}
+
+            const { globby } = await import("${{ github.workspace }}/node_modules/globby/index.js");
+            const toml = require('toml');
+            const fs = require('fs');
+            const axios = require('axios');
+            const { spawn } = require('child_process');
+
+            async function get_crate_details(name) {
+              try {
+                return (await axios(`https://crates.io/api/v1/crates/${name}`)).data
+              } catch(error) {
+                console.error("Error fetching crate details:", error)
+              }
+            }
+
+            function run_command_stream(command_str) {
+              const [command, ...args] = command_str.split(" ")
```

Comment:
danger of shell quoting problems here. would it be so bad to have to call `run_command_stream(["foo", "bar"])`?

Reply:
Good point. Let me change 🙏